lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -350,7 +350,78 @@ def disable(self) -> OpenSearchPluginConfig:
             config_entries_to_add={"knn.plugin.enabled": False},
         )
 
+    @property
+    def dependencies(self) -> Optional[List[str]]:
+        """Returns a list of plugin name dependencies."""
+        return ["opensearch-ml"]
+
     @property
     def name(self) -> str:
         """Returns the name of the plugin."""
         return "opensearch-knn"
+
+
+class OpenSearchPluginMLCommons(OpenSearchPlugin):
+    """Implements the opensearch-ml-commons plugin."""
+
+    def config(self) -> OpenSearchPluginConfig:
+        """Returns a plugin config object to be applied for enabling the current plugin."""
+        return OpenSearchPluginConfig(
+            config_entries_to_add={
+                "plugins.ml_commons.only_run_on_ml_node": True,
```

Comment:
all these seem to be populating the default values of the plugin, it should be safe to remove